guides/micronaut-k8s/orders/java/src/main/java/example/micronaut/auth/CredentialsChecker.java

```diff
@@ -0,0 +1,35 @@
+package example.micronaut.auth;
+
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.security.authentication.AuthenticationProvider;
+import io.micronaut.security.authentication.AuthenticationRequest;
+import io.micronaut.security.authentication.AuthenticationResponse;
+import jakarta.inject.Singleton;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.FluxSink;
+
+@Singleton
+public class CredentialsChecker implements AuthenticationProvider {
+
+    private final Credentials credentials;
+
+    public CredentialsChecker(Credentials credentials) {
+        this.credentials = credentials;
+    }
+
+    @Override
+    public Publisher<AuthenticationResponse> authenticate(@Nullable HttpRequest<?> httpRequest,
+                                                          AuthenticationRequest<?, ?> authenticationRequest) {
+        return Flux.create(emitter -> {
```

Comment:
This should be a `Mono`. See https://micronaut-projects.github.io/micronaut-security/snapshot/guide/#authenticationProviders for an example.